city_scrapers/spiders/il_pollution_control.py

```diff
@@ -0,0 +1,160 @@
+                if dt is None:
+                    continue  # Could not find a matching format_str - can't process link.
+
+                self.link_map[dt] = url
+
+    def _parse_json(self, response):
+        """ Parse JSON from https://pcb.illinois.gov/ClerksOffice/GetCalendarEvents -> Meetings """
+        data = json.loads(response.body_as_unicode())
+
+        for item in data:
+            if any(
+                s in item["CalendarTypeDesc"].lower() for s in ("holiday", "seminar", "hearing")
+            ):
+                continue  # Not interested in this event type
+
+            title = item["CalendarTypeDesc"].replace("CANCELLED", "").strip()
+            meeting = Meeting(
+                title=title,
+                description="",  # Too inconsistent to parse accurately
+                classification=self._parse_classification(title),
+                start=self._parse_start(item),
+                end=None,
+                all_day=item["IsFullDay"],
+                time_notes="",
+                location=self._parse_location(item),
+                links=list(),
+                source=self._parse_source(item),
+            )
+
+            meeting["links"] = self._parse_links(meeting)
+            meeting["status"] = self._get_status(
+                meeting, text=" ".join([item['CalendarTypeDesc'], item['Description']]).lower()
+            )
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_classification(self, title):
+        """Parse or generate classification from allowed options."""
+        if "Board" in title:
+            return BOARD
+        else:
+            return NOT_CLASSIFIED
+
+    def _parse_start(self, item):
+        return datetime.strptime(item["StartDateTime"], '%m/%d/%Y %I:%M:%S %p')
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        text = " ".join([item['Description'], item['Location']]).lower()
+        if "thompson" in text:
+            return {
+                "address":
+                    "James R. Thompson Center - 100 W. Randolph St. Suite 11-500, Chicago, IL 60601",  # noqa
+                "name": "Chicago IPCB Office",
+            }
+        elif "springfield" in text or "llinois pollution control board" in text:
+            return {
+                "address": "1021 N. Grand Ave. E. - Room 1244 N, Springfield, IL 62702",
+                "name": "Springfield IPCB Office",
+            }
+        elif "sangamo room" in text:
+            return {
+                "address": "1021 N. Grand Ave. E. - Sangamo Room, Springfield, IL 62702",
+                "name": "Illinois EPA",
+            }
+        else:
+            return {
+                "address": "",
+                "name": "",
+            }
+
+    def _parse_links(self, meeting):
+        """ Associate Meeting objects with previously-scraped links """
+        key = meeting['start'].date()
+        if key in self.link_map:
```

Comment:
The tests just made me notice that this needs to return an empty list instead of `None` if nothing is found

Reply:
Whoops, alrighty, I'll get that in there.